### DISCUSSION TITLE
saltstack onedir minion to old saltstack master

### DISCUSSION THREAD

**Discussion by bradawk1** [saltstack/salt#66972] [General]
Our salt masters are running on RHEL 8 and are at version 3006.7.  We have a few test RHEL 9 servers.  The only 3006 version of onedir I have available is 3006.9.  Do you think that will be OK with the 3006.7 master?

**Comment by dafyddj** [comment]
Probably. Can you try one minion to see?

**Reply by bradawk1** [reply]
I will.  My servers are not connected to the Internet.  I'm just waiting on the team that pulls in this data to grab the GPG keys.  Not sure why they forgot them?  :)

**Comment by bradawk1** [comment]
I removed the old salt and dependencies and then installed salt onedir 3006.9 salt and salt-minion rpms.  On all four systems that I tried, the installation caused the servers to crash and reboot.  After the servers came back up, I was able to run a dnf history redo on each and it completed without any problems.  In general, all of the problems I was seeing with the previous salt version are gone, but I am running one state file that has several file.managed stanzas and keep getting:

DeprecationWarning: Please stop importing 'salt.utils.psutil_compat' and instead import 'psutil' directly as there's no longer a need for a compatability layer.    - I am investigating this one.
WARNING source_hash is only used with 'http', 'https', or 'ftp'  - never got this before.  We are using gitfs and source is giving as salt://somedir...
ERROR Unable to manage file: File client timed out after 180 seconds.  - got this a lot.  I think I might have to run in debug mode to try and understand which files it is having trouble with.

So, some progress and some more investigations.  :)

**Comment by dmurphy18** [comment]
@bradawk1 
```
DeprecationWarning: Please stop importing 'salt.utils.psutil_compat' and instead import 'psutil' directly as there's no longer a need for a compatability layer. - I am investigating this one.
```
This has already been taken care of and will be removed in the next release

**Comment by bradawk1** [comment]
Thanks!  I won't worry about that one.  :)

**Comment by bradawk1** [comment]
I can report that the 3006.9 salt minion does not work with the 3006.7 salt master.  I kept getting all sorts of errors.  I then found onedir 3006.7 salt rpms.  I removed the 3006.9 and installed those.  The salt states all ran with no problems.
